Allows the TTY name to be passed into argv[2]. It still defaults to
/dev/tty. This is necessary because that doesn't work when the tty
is mapped to /dev/console, a real serial terminal is required.

--- busybox-1.1.3/miscutils/rx.c~	2006-03-22 22:16:24.000000000 +0100
+++ busybox-1.1.3/miscutils/rx.c	2017-02-03 16:08:50.574235177 +0100
@@ -285,11 +285,11 @@
 	int n;
 	char error_buf[256];
 
-	if (argc != 2)
+	if (argc < 2 || argc > 3)
 			bb_show_usage();
 
 	fn = argv[1];
-	ttyfd = open("/dev/tty", O_RDWR);
+	ttyfd = open(argc > 2 ? argv[2] : "/dev/tty", O_RDWR);
 	if (ttyfd < 0)
 			bb_error_msg_and_die("%s: open on /dev/tty failed: %m\n", argv[0]);
 
